services/datamanager/datasync/upload_arbitrary_file.go

```diff
@@ -41,17 +41,14 @@ func uploadArbitraryFile(ctx context.Context, client v1.DataSyncServiceClient, f
 		return errors.Wrapf(err, "error syncing %s", f.Name())
 	}
 
-	_, err = stream.CloseAndRecv()
-	if err != nil {
+	if _, err := stream.CloseAndRecv(); err != nil {
 		return errors.Wrapf(err, "received error response while syncing %s", f.Name())
 	}
 
 	return nil
 }
 
 func sendFileUploadRequests(ctx context.Context, stream v1.DataSyncService_FileUploadClient, f *os.File) error {
-	//nolint:errcheck
-	defer stream.CloseSend()
```

Comment:
Why is this removed? Is this in order to synchronously have  `uploadArbitraryFile` and `uploadDataCaptureFile` in charge of closing the channel and getting the client response (via `CloseAndRecv`), rather than splitting that between the helper and main loop?

Reply:
`CloseAndRecv` is the recommended/intended method for ending a client side stream ([docs](https://grpc.io/docs/languages/go/basics/#client-side-streaming-rpc-1). I believe this is because the stream ending and a response being received are supposed to always happen in this fixed order (since there is only one response message). In bidi streams where these can be interleaveed, [CloseSend](https://grpc.io/docs/languages/go/basics/#bidirectional-streaming-rpc-1) is used